### PR TITLE
🎉 Source Mysql - added basic MySql performance tests

### LINKED ISSUE #7489: Source MySQL: stress test 1000 tables / columns per sync
## Tell us about the problem you're trying to solve
As part of making our databases more reliable, we should start working on a test suite containing benchmarks for validating that our DB connectors are able to serve certain workloads/scales. 

The idea is that today, we see a lot of issues around DB replication that we cannot reproduce (doc containing [example of problems](https://docs.google.com/spreadsheets/d/1E-fo4EE4pTAJFz3Kfsa347PjAir22CsYpohaYPieVSo/edit#gid=0)) so we want to create a test harness specifically for reproducing this style of test. 

## Describe the solution you’d like

- [ ] Create Gradle sub-project to run stress test independently
- [ ] Create stress test including 1000 streams & 1000 columns

_Note: test infrastructure should not depend on MySQL connector_

## Describe the alternative you’ve considered or used
A clear and concise description of any alternative solutions or features you've considered or are using today.

## Additional context
The solution should be based on the described [issue](https://github.com/airbytehq/airbyte/issues/5754#issue-983759059)

## Are you willing to submit a PR?
Yes

### PR DESCRIPTION
**Pull request by etsybaev**
## What
We want to perform more performance tests for connectors.

## How
Added skeleton and base tests for the Source MySQL connector.
Since now we would be able to run performance tests locally or by leaving a PR's comment.
Locally: "./gradlew clean :airbyte-integrations:connectors:source-mysql:performanceTest"
In comment: "/test-performance connector=connectors/source-mysql"

As preconditions more scripts were added https://github.com/airbytehq/airbyte/pull/7920

## Recommended reading order
1. `x.java`
2. `y.python`

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Code reviews completed
- [x] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [x] Create a non-forked branch based on this PR and test the below items on it
- [x] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [x] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

**Review by tuliren — APPROVED** [review]
Looks good!

One general suggestion is to use jOOQ code to construct the queries. In that way, many of the string templates can be omitted. But this is not a high priority requirement. Only do it if you think it helps.

File: airbyte-integrations/bases/standard-source-test/src/main/java/io/airbyte/integrations/standardtest/source/AbstractSourcePerformanceTest.java [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+  /**
+   * Get a test field'stype that will be used in DB for table creation.
+   *
+   * @return a test's field type. Ex: varchar(8)
+   */
+  protected abstract String getTestFieldType();
+
+  @Override
+  protected void setupEnvironment(final TestDestinationEnv environment) throws Exception {
+    // DO NOTHING. Mandatory to override. DB will be setup as part of each test
```
[the remark this answers is not in the document]
Comment (etsybaev) [reply airbyte-integrations/bases/standard-source-test/src/main/java/io/airbyte/integrations/standardtest/source/AbstractSourcePerformanceTest.java:107]: Hi Liren. This method is abstract and supposed to be called before tests for DB preparation. Meanwhile, we do it individually for each test as part of test itself. So if I remove this override  - I see such an error.  Or maybe I'm missing something... Thanks 
Screenshot from 2021-11-12 11-41-39

File: airbyte-integrations/connectors/source-mysql/src/test-performance/java/io/airbyte/integrations/source/mysql/MySqlSourcePerformanceTest.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+            config.get("database").asText()),
+        "com.mysql.cj.jdbc.Driver",
+        SQLDialect.MYSQL,
+        "zeroDateTimeBehavior=convertToNull");
+
+    super.databaseName = container.getDatabaseName();
+
+    // It disable strict mode in the DB and allows to insert specific values.
+    // For example, it's possible to insert date with zero values "2021-00-00"
+    database.query(ctx -> ctx.fetch("SET @@sql_mode=''"));
```
[the remark this answers is not in the document]
Comment (etsybaev) [reply airbyte-integrations/connectors/source-mysql/src/test-performance/java/io/airbyte/integrations/source/mysql/MySqlSourcePerformanceTest.java]: Done, thanks!

**Comment by tuliren** [comment]
@etsybaev, the slash command dispatch only works if the command has already been registered in the master branch. You can see the dispatch log [here](https://github.com/airbytehq/airbyte/runs/4192693395?check_suite_focus=true):

> Using configuration from yaml inputs.
> ##[debug]Commands: [ 'test', 'publish', 'publish-cdk', 'gke-kube-test' ]
> ...
> Command 'test-performance' is not registered for dispatch.

This is confirmed with [this](https://github.com/peter-evans/slash-command-dispatch/issues/100#issuecomment-801325833) comment in the `slash-command-dispatch` repo.

So you can create a separate PR to merge that new command to master first.

File: airbyte-integrations/bases/standard-source-test/src/main/java/io/airbyte/integrations/standardtest/source/AbstractSourcePerformanceTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (tuliren) [reply airbyte-integrations/bases/standard-source-test/src/main/java/io/airbyte/integrations/standardtest/source/AbstractSourcePerformanceTest.java:107]: Oh, I misunderstood the comment. I thought the "mandatory to override" meant any class extending from this abstract class should override it. Never mind.

**Review by alexandr-shegeda — APPROVED** [review]
nice work!

**Pull request merged**